brick-editor.js

```diff
@@ -97,43 +99,39 @@ function buttonHandler(i) { // eslint-disable-line no-unused-vars
  *
  * The setTimeout() function is necessary to allow highlighting before confirm dialog popup
  *
+ * @param {string} buffer - The current editor text.
  * @param {AST} ast - The root of the ast to search through.
  * @param {Location} selection - An object with start/end lineNumber and start/end column properties.
  * @returns {undefined}
  */
-function selectionBranch(ast, selection) {
+function selectionBranch(buffer, ast, selection) {
     var node = deleteSelected(ast, selection);
-    highlight(node.loc.start.line, node.loc.start.column, node.loc.end.line, node.loc.end.column);
-    setTimeout(function () {
-        var response = confirm("Are you sure you wish to delete?");
-        if (response) {
-            editor.setValue(deleteBlock(ast, node));
-        } else {
-            unhighlight();
-        }
-    }, 100);
+    if (highlighted) {
+        setValue(buffer, deleteBlock(ast, node));
+    } else {
+        highlight(node.loc.start.line, node.loc.start.column, node.loc.end.line, node.loc.end.column);
+        highlighted = true;
```

Comment:
I would put `highlighted = true` in `highlight` and `highlight = false` in `unhighlight()`, so those are the only places where the global variable is set.